lib/pbench/test/unit/server/test_upload.py

```diff
@@ -139,12 +143,15 @@ def test_missing_filename_extension(
         """Test with URL uploading a file named "f" which is missing the
         required filename extension"""
         expected_message = "File extension not supported, must be .tar.xz"
-        response = client.put(
-            f"{server_config.rest_uri}/upload/f",
-            headers={
-                "Authorization": "Bearer " + pbench_drb_token,
-            },
-        )
+        with BytesIO(b"junk") as f:
```

Comment:
> I'm not entirely certain that simply letting it go out of scope will necessarily free the buffer

I am...otherwise that would be a memory leak!...although, Python makes no guarantees about the timeliness of destructor execution....

> An explicit close is better, and the context manager is a clean way to manage that.

As you wish....  😉